packages/apollo-gateway/src/__tests__/queryPlanCucumber.test.ts

```diff
@@ -0,0 +1,138 @@
+import { composeServices, buildFederatedSchema, normalizeTypeDefs } from '@apollo/federation';
+
+import { QueryPlan } from '../..';
+import { LocalGraphQLDataSource } from '../datasources/LocalGraphQLDataSource';
+import { buildQueryPlan, buildOperationContext } from '../buildQueryPlan';
+
+const buildQueryPlanFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/build-query-plan.feature'
+);
+
+
+const features = [
+  buildQueryPlanFeature
+];
+
+function buildLocalService(modules: GraphQLSchemaModule[]) {
+  const schema = buildFederatedSchema(modules);
+  return new LocalGraphQLDataSource(schema);
+}
+
+features.forEach((feature) => {
+  defineFeature(feature, (test) => {
+    feature.scenarios.forEach((scenario) => {
+      test(scenario.title, async ({ given, when, then }) => {
+        let query: DocumentNode;
+        let schema: GraphQLSchema;
+        let errors: GraphQLError[];
+        let queryPlan: QueryPlan;
+        let options: any = {};
+
+        const serviceMap = Object.fromEntries(
+          ['accounts', 'product', 'inventory', 'reviews', 'books', 'documents'].map(
+            serviceName => {
+              return [
+                serviceName,
+                buildLocalService([
+                  require(path.join(
+                    __dirname,
+                    '__fixtures__/schemas',
+                    serviceName,
+                  )),
+                ]),
+              ] as [string, LocalGraphQLDataSource];
+            },
+          ),
+        );
+
+        ({ schema, errors } = composeServices(
+          Object.entries(serviceMap).map(([serviceName, service]) => ({
+            name: serviceName,
+            typeDefs: normalizeTypeDefs(service.sdl()),
+          })),
+        ));
+
+        if (errors && errors.length > 0) {
+          throw new GraphQLSchemaValidationError(errors);
+        }
+
+        const givenQuery = () => {
+          given(/^query$/im, (operation) => {
+            query = gql(operation)
+          })
+        }
+
+        const whenUsingAutoFragmentization = () => {
+          when(/using autofragmentization/i, () => {
+            options = { autoFragmentization: true };
+          })
+        }
+
+        const thenQueryPlanShouldBe = () => {
+          then(/^query plan$/i, (expectedQueryPlan) => {
+            queryPlan = buildQueryPlan(
+              buildOperationContext(schema, query, undefined),
+              options
+            );
+
+            const serializedPlan = JSON.parse(JSON.stringify(queryPlan, serializeQueryPlanNode));
+            const parsedExpectedPlan = JSON.parse(expectedQueryPlan);
+
+            expect(serializedPlan).toEqual(parsedExpectedPlan);
+          })
+        }
+
+        // step over each defined step in the .feature and execute the correct
+        // matching step fn defined above
+        scenario.steps.forEach(({ stepText }) => {
+          if (/^query$/i.test(stepText)) givenQuery();
```

Comment:
Not asking for a change, but more a better understanding: Could we just use string equality, or do we expect `QuErY`?  (And if we do expect the latter, perhaps we could just `stepText.toLowerCase()` it once?

```suggestion
          if (stepText === "query") givenQuery();
```

Reply:
Yeah. this was more based on the pattern of the more complicated step names in the integration test suite, where there could be arguments in the step title. For these, we can definitely simplify! :)